MailNotificationServiceExtension/NotificationService.swift

```diff
@@ -25,7 +25,12 @@ import MailResources
 import RealmSwift
 import UserNotifications
 
-class NotificationService: UNNotificationServiceExtension {
+final class NotificationService: UNNotificationServiceExtension {
+    /// Making sure the DI is registered at a very early stage of the app launch.
+    private let dependencyInjectionHook = EarlyDIHook()
```

Comment:
You are correct, I think this is left code from automatic merge since a lot was refactored.

I cleaned it, making sure in the process that it all was corectly initiated in `EarlyDIHook`